### PR TITLE
task 08 solved

### PR DESCRIPTION
**Pull request by arthur-borisenko**

### CODE REVIEW

File: week6/08-prokhodnoi-ball/task.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    for el in input_list:
+        if key(el) == pattern:
+            res += 1
+    return res
+
+
+def get_min_score(input_lines, k):
+    students = []
+    for line in input_lines:
+        student = Student()
```
Comment (IgorNB) [inline week6/08-prokhodnoi-ball/task.py:28]: extract string to class method parse_student

File: week6/08-prokhodnoi-ball/task.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        temp_student_data = line.split()
+        student.third, student.second, student.first = \
+            map(int, temp_student_data[-1:-4:-1])
+        student.sum = student.third + student.first + student.second
+        student.name = " ".join(temp_student_data[:-3])
+        if (student.first >= 40 and student.second >= 40 and
```
Comment (IgorNB) [inline week6/08-prokhodnoi-ball/task.py:34]: extract condition to check_min_requirements

File: week6/08-prokhodnoi-ball/task.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+def main():
+    with open("input.txt", encoding="utf-8") as input_file, open(
+            "output.txt", "w", encoding="utf-8") as output_file:
+        k = int(input_file.readline())
+        input_lines = input_file.readlines()
+        print(get_min_score(input_lines, k), file=output_file)
+
+
+def count_by_key(input_list, pattern, key=lambda x: x):
```
Comment (IgorNB) [inline week6/08-prokhodnoi-ball/task.py:17]: def count(input_list, key=lambda x: True):
    res = 0
    for el in input_list:
        if key(el):
            res += 1
    return res


and usage

 count_max_total = count(students, key=lambda el: el.total == max_total)

File: week6/08-prokhodnoi-ball/task.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+            res += 1
+    return res
+
+
+def get_min_score(input_lines, k):
+    students = []
+    for line in input_lines:
+        student = Student()
+        temp_student_data = line.split()
+        student.third, student.second, student.first = \
```
Comment (IgorNB) [inline week6/08-prokhodnoi-ball/task.py:30]: student.thirdScore, student.secondScore, student.firstScore = \
        int(words[-1]), int(words[-2]), int(words[-3]) is simpler

File: week6/08-prokhodnoi-ball/task.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+                student.third >= 40):
+            students.append(student)
+    students.sort(key=lambda student: student.sum, reverse=True)
+    if len(students) <= k:
+        return 0
+    if count_by_key(students,
+                    max(students, key=lambda el: el.sum).sum,
+                    lambda x: x.sum) > k:
+        return 1
+    else:
```
Comment (IgorNB) [inline week6/08-prokhodnoi-ball/task.py:44]: else not needed - return in prev if

File: week6/08-prokhodnoi-ball/task.py [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        enrollment_score = students[0].sum
+        fail_score = students[k].sum
+        for i in range(k):
```
Comment (IgorNB) [inline week6/08-prokhodnoi-ball/task.py:47]: extract method find_min_total_greater_then

usage return find_min_total_greater_then(students, students[k].total)

File: week6/08-prokhodnoi-ball/task.py [hunk 7]
```diff
@@ -0,0 +1,54 @@
+class Student:
+    name = ""
```
Comment (IgorNB) [inline week6/08-prokhodnoi-ball/task.py:2]: fullName

File: week6/08-prokhodnoi-ball/task.py [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    first = 0
```
Comment (IgorNB) [inline week6/08-prokhodnoi-ball/task.py:3]: firstScore

File: week6/08-prokhodnoi-ball/task.py [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+    second = 0
+    third = 0
+    sum = 0
```
Comment (IgorNB) [inline week6/08-prokhodnoi-ball/task.py:6]: total

File: week6/08-prokhodnoi-ball/task.py [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+
+def count_by_key(input_list, pattern, key=lambda x: x):
+    res = 0
+    for el in input_list:
+        if key(el) == pattern:
+            res += 1
+    return res
+
+
+def get_min_score(input_lines, k):
```
Comment (IgorNB) [inline week6/08-prokhodnoi-ball/task.py:25]: rename to find_enrollment_score(students, k)

usage

def find_enrollment_score(students, k):
    students = list(filter(lambda el: check_min_requirements(el), students))
    students.sort(key=lambda el: el.total, reverse=True)

    if len(students) <= k:
        return 0

    max_total = max(students, key=lambda el: el.total).total
    count_max_total = count(students, key=lambda el: el.total == max_total)
    if count_max_total > k:
        return 1

    return find_min_total_greater_then(students, students[k].total)

**Review by IgorNB — CHANGES_REQUESTED** [review]

File: week6/08-prokhodnoi-ball/task.py (hunk 10 above)
Comment (arthur-borisenko) [reply week6/08-prokhodnoi-ball/task.py]: done

File: week6/08-prokhodnoi-ball/task.py (hunk 1 above)
Comment (arthur-borisenko) [reply week6/08-prokhodnoi-ball/task.py]: done

File: week6/08-prokhodnoi-ball/task.py (hunk 4 above)
Comment (arthur-borisenko) [reply week6/08-prokhodnoi-ball/task.py]: done

File: week6/08-prokhodnoi-ball/task.py (hunk 2 above)
Comment (arthur-borisenko) [reply week6/08-prokhodnoi-ball/task.py]: done

File: week6/08-prokhodnoi-ball/task.py (hunk 5 above)
Comment (arthur-borisenko) [reply week6/08-prokhodnoi-ball/task.py]: done

File: week6/08-prokhodnoi-ball/task.py (hunk 6 above)
Comment (arthur-borisenko) [reply week6/08-prokhodnoi-ball/task.py]: done

File: week6/08-prokhodnoi-ball/task.py (hunk 3 above)
Comment (arthur-borisenko) [reply week6/08-prokhodnoi-ball/task.py]: done

File: week6/08-prokhodnoi-ball/task.py (hunk 7 above)
Comment (arthur-borisenko) [reply week6/08-prokhodnoi-ball/task.py]: done

File: week6/08-prokhodnoi-ball/task.py (hunk 9 above)
Comment (arthur-borisenko) [reply week6/08-prokhodnoi-ball/task.py]: done

File: week6/08-prokhodnoi-ball/task.py (hunk 8 above)
Comment (arthur-borisenko) [reply week6/08-prokhodnoi-ball/task.py]: done

**Review by IgorNB — APPROVED** [review]

**Pull request merged**